### PR TITLE
Fix #621, update action versions to v3

### LINKED ISSUE #621: Warning about node.js 12 deprecation in github actions
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I performed a cursory search to see if the bug report is relevant, not redundant, nor in conflict with other tickets.

**Describe the bug**
Github produces a warning about node.js 12 being obsolete with every workflow run.  It has been fixed upstream, but CFS uses the old version.  This can be fixed by updating the version of the action from v2 to v3.

**To Reproduce**
Run any of the reusable github workflows that use the "v2" actions.

**Expected behavior**
Should not use deprecated versions of actions

**System observed on:**
Github hosted runner

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

### PR DESCRIPTION
**Pull request by jphickey**
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I signed and emailed the appropriate [Contributor License Agreement](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
Update cache, checkout, and upload-artifact actions from v2 to v3. Should be a drop-in replacement and removes a warning about node.js 12 deprecation.

Fixes #621

**Testing performed**
Run all github workflows

**Expected behavior changes**
No more warnings about node.js 12 deprecation

**System(s) tested on**
Github hosted runner

**Contributor Info - All information REQUIRED for consideration of pull request**
Joseph Hickey, Vantage Systems, Inc.

### CODE REVIEW

**Comment by ArielSAdamsNASA** [comment]
@jphickey Looks good, but I don't see where the node.js warning is removed. For example, click the Format Check workflow triggered by this PR and take a look at the workflow file. It uses v3, but the warning is still there. Here is a link: https://github.com/nasa/cFS/actions/runs/3535753791

**Comment by jphickey** [comment]
> @jphickey Looks good, but I don't see where the node.js warning is removed. For example, click the Format Check workflow triggered by this PR and take a look at the workflow file. It uses v3, but the warning is still there. Here is a link: https://github.com/nasa/cFS/actions/runs/3535753791

Thanks for catching that, it looks like the warning in that particular run was from the 3rd-party action "gsactions/commit-message-checker@v1", not the github actions that were updated in this PR.  

We'll need to see if there is an update/new version from the original author.  I'll check that.  I'd still recommend merging this PR, as it fixes _most_ of them - there may be some left but I can take care of those as they appear.

**Comment by jphickey** [comment]
I checked and yes, the "commit-message-checker" does have a version 2 now.  So I pushed another commit (see 5a1174f) that updates this action from v1 to v2 and the warning is gone.  See https://github.com/nasa/cFS/actions/runs/3593359346.

We will just have to keep an eye out for remaining deprecation warnings and watch for new versions of whatever action is triggering them.  Problem is when there are a lot, things get buried, so even if this may not fix 100% of the warnings, it will make the remaining ones easier to spot and fix.

**Review by ArielSAdamsNASA — APPROVED** [review]

File: .github/workflows/build-cfs-deprecated.yml [hunk 1]
```diff
@@ -162,16 +162,16 @@ jobs:
...
     strategy:
       fail-fast: false
       matrix:
         buildtype: [debug, release]
 
-    steps:      
+    steps:
       - name: Cache Source and Deprecated Build
         id: cache-src-bld
-        uses: actions/cache@v2
```
Comment (tiffanydawnatkins-cloud) [inline .github/workflows/build-cfs-deprecated.yml:174]: actions/cache@v3

File: .github/workflows/build-cfs-deprecated.yml [hunk 2]
```diff
@@ -188,34 +188,34 @@ jobs:
...
             echo "Waiting for CFE Tests"
             sleep 60
           done
-          
+
           ../host/cmdUtil --endian=LE --pktid=0x1806 --cmdcode=2 --half=0x0002
         working-directory: ./build/exe/cpu1/
 
       - name: Archive Functional Test Artifacts
-        uses: actions/upload-artifact@v2
```
Comment (tiffanydawnatkins-cloud) [inline .github/workflows/build-cfs-deprecated.yml:208]: uses:%20actions/upload-artifact@v3

File: .github/workflows/build-cfs-rtems4.11.yml [hunk 3]
```diff
@@ -124,7 +124,7 @@ jobs:
 
       # Always archive test logs
       - name: Archive cFS Test Artifacts
-        uses: actions/upload-artifact@v2
```
Comment (tiffanydawnatkins-cloud) [inline .github/workflows/build-cfs-rtems4.11.yml:127]: uses:%20actions/upload-artifact@v3

File: .github/workflows/build-cfs-rtems4.11.yml [hunk 4]
```diff
@@ -92,7 +92,7 @@ jobs:
     steps:
       # Checks out a copy of your repository on the ubuntu-latest machine
       - name: Checkout code
-        uses: actions/checkout@v2
```
Comment (tiffanydawnatkins-cloud) [inline .github/workflows/build-cfs-rtems4.11.yml:95]: uses: actions/checkout@v3

File: .github/workflows/build-cfs-rtems4.11.yml [hunk 5]
```diff
@@ -51,7 +51,7 @@ jobs:
     steps:
       # Check out the cfs bundle
       - name: Checkout code
-        uses: actions/checkout@v2
```
Comment (tiffanydawnatkins-cloud) [inline .github/workflows/build-cfs-rtems4.11.yml:54]: uses: actions/checkout@v3

File: .github/workflows/build-cfs.yml [hunk 6]
```diff
@@ -45,35 +45,35 @@ jobs:
...
     # Set the type of machine to run on
     env:
-      BUILDTYPE: ${{ matrix.buildtype }}    
+      BUILDTYPE: ${{ matrix.buildtype }}
 
     steps:
       - name: Cache Source and Build
         id: cache-src-bld
-        uses: actions/cache@v2
+        uses: actions/cache@v3
```
Comment (tiffanydawnatkins-cloud) [inline .github/workflows/build-cfs.yml:53]: https://github.com/nasa/cFS/pull/622/changes#diff-16341bd5d8524a30d83ad6a548bb54be76c8b95488301f44a94a376f71aff5c5L51

File: .github/workflows/build-deploy-doc.yml
Comment (tiffanydawnatkins-cloud) [inline .github/workflows/build-deploy-doc.yml:69]: password fail

**Pull request merged**